app/controllers/partners/providers_controller.rb

```diff
@@ -68,19 +68,22 @@ def user_params
   end
 
   def find_account
-    @account = @partner.providers.find(params[:id])
+    @account = @partner.providers.find(permitted_params[:id])
   end
 
   def application_plans
     @application_plans ||= @partner.application_plans
   end
 
   def selected_plan
-    if @selected_plan = application_plans.find_by_system_name(params[:application_plan])
+    if @selected_plan = application_plans.find_by(system_name: permitted_params[:application_plan])
       @selected_plan
     else
       application_plans.first
     end
   end
 
+  def permitted_params
+    params.permit!.to_h
+  end
```

Comment:
We should list all needed params instead

Reply:
@hallelujah changes done